back/lib/tasks/single_use/20240910_migrate_initiatives_to_proposals.rake

```diff
@@ -0,0 +1,811 @@
+namespace :initiatives_to_proposals do
+  desc 'Replace the Citizenlab default moderator with Go Vocal attributes.'
+  task :migrate_proposals, [] => [:environment] do
+    reporter = ScriptReporter.new
+    Tenant.safe_switch_each do |tenant|
+      puts "Migrating initiatives to proposals for #{tenant.host}"
+      next if !AppConfiguration.instance.feature_activated?('initiatives') && !Initiative.exists?
+
+      SettingsService.new.activate_feature!('input_cosponsorship') if AppConfiguration.instance.feature_activated?('initiative_cosponsors')
+
+      project = rake_20240910_create_proposals_project(reporter)
+      next if !project
+
+      rake_20240910_substitute_homepage_element(project, reporter)
+      rake_20240910_replace_navbaritem(project, reporter)
+      rake_20240910_migrate_input_statuses(reporter)
+      Initiative.where.not(publication_status: 'draft').each do |initiative|
+        proposal_attributes = rake_20240910_proposal_attributes(initiative, project)
+        proposal = Idea.new proposal_attributes
+        next if !rake_20240910_assign_idea_status(proposal, initiative, reporter)
```

Comment:
This should never happen, but if some issue happened, it is logged in the error reporter and reported together with all other potential issues. Why I think it's better not to fail:
- We can list multiple errors in one (long) test run, so we can fix multiple errors and save time compared to having to execute a test run for each new error.
- If for some reason we would still have an error on production (we shouldn't, but you never know), the script wouldn't crash and would keep on migrating the other tenants.

That being said, the testing on copies of the databases is not going smoothly because it's always crashing due to connection errors and this is preventing me from seeing possible errors (currently working on improving this part).